node/src/components/linear_chain/signature_cache.rs

```diff
@@ -40,28 +40,16 @@ impl SignatureCache {
         }
     }
 
-    /// Get signatures from the cache to be updated.
-    /// If there are no signatures, create an empty signature to be updated.
-    pub(super) fn get_known_signatures(
-        &self,
-        block_hash: &BlockHash,
-        block_era: EraId,
-    ) -> BlockSignatures {
-        match self.signatures.get(block_hash) {
-            Some(signatures) => signatures.clone(),
-            None => BlockSignatures::new(*block_hash, block_era),
-        }
-    }
-
     /// Returns whether finality signature is known already.
     pub(super) fn known_signature(&self, fs: &FinalitySignature) -> bool {
         let FinalitySignature {
             block_hash,
             public_key,
+            signature,
             ..
         } = fs;
         self.signatures
             .get(block_hash)
-            .map_or(false, |bs| bs.has_proof(public_key))
+            .map_or(false, |bs| bs.has_proof(public_key, signature))
```

Comment:
I don't think the signature itself should matter.

Reply:
Done in [`f5f6038` (#1268)](https://github.com/CasperLabs/casper-node/pull/1268/commits/f5f603830d33577186c4c7072cbb578714d8e773)